tests/src/test/java/alluxio/client/AbstractFileOutStreamIntegrationTest.java

```diff
@@ -62,27 +57,76 @@ public void before() throws Exception {
   }
 
   /**
+   * Helper to write an Alluxio file with stream of bytes of increasing byte value.
+   *
+   * @param filePath path of the tmp file
+   * @param fileLen length of the file
+   * @param op options to create file
+   * @throws Exception if error occurs
+   */
+  protected void writeIncreasingBytesToFile(AlluxioURI filePath, int fileLen, CreateFileOptions op)
+      throws Exception {
+    FileOutStream os = mFileSystem.createFile(filePath, op);
+    for (int k = 0; k < fileLen; k++) {
+      os.write((byte) k);
+    }
+    os.close();
+  }
+
+  /**
+   * Helper to write an Alluxio file with increasing byte array.
+   *
+   * @param filePath path of the tmp file
+   * @param fileLen length of the file
+   * @param op options to create file
+   * @throws Exception if error occurs
+   */
+  protected void writeIncreasingByteArrayToFile(AlluxioURI filePath, int fileLen,
+      CreateFileOptions op) throws Exception {
+    FileOutStream os = mFileSystem.createFile(filePath, op);
+    os.write(BufferUtils.getIncreasingByteArray(fileLen));
+    os.close();
+  }
+
+  /**
+   * Helper to write an Alluxio file with two increasing byte arrays separately.
+   *
+   * @param filePath path of the tmp file
+   * @param fileLen length of the file
+   * @param op options to create file
+   * @throws Exception if error occurs
+   */
+  protected void writeTwoIncreasingByteArraysToFile(AlluxioURI filePath, int fileLen,
+      CreateFileOptions op) throws Exception {
+    FileOutStream os = mFileSystem.createFile(filePath, op);
+    os.write(BufferUtils.getIncreasingByteArray(0, fileLen / 2), 0, fileLen / 2);
+    os.write(BufferUtils.getIncreasingByteArray(fileLen / 2, fileLen / 2), 0, fileLen / 2);
+    os.close();
+  }
+
+  /**
    * Checks that we wrote the file correctly by reading it every possible way.
    *
    * @param filePath path of the tmp file
-   * @param underStorageType type of under storage write
+   * @param checkAlluxioStorage whether to check this file in Alluxio storage
+   * @param checkUnderStorage whether to check this file in under storage
    * @param fileLen length of the file
-   * @param increasingByteArrayLen expected length of increasing bytes written in the file
+   * @throws Exception if error occurs
    */
-  protected void checkWrite(AlluxioURI filePath, UnderStorageType underStorageType, int fileLen,
-      int increasingByteArrayLen) throws Exception {
-    for (CreateFileOptions op : getOptionSet()) {
-      URIStatus status = mFileSystem.getStatus(filePath);
+  protected void checkFile(AlluxioURI filePath, boolean checkAlluxioStorage,
```

Comment:
The invocations of this method would be easier to understand if we break it into `checkFileInAlluxio` and `checkFileInUnderStorage`. Right now they are magical booleans